### PR TITLE
Add `--verbose` flag

### LINKED ISSUE #38: show more details in results
It would be handy if the results showed more details (similar to the fast.com website) such as the latency stats and the server location.

### PR DESCRIPTION
**Pull request by itaisteinherz**
Resolves #38 (and will close #40).

Screenshot

### CODE REVIEW

**Comment by itaisteinherz** [comment]
I just noticed #40 😅 (I worked on this for a few days now and only got to push it now). Sorry for not claiming the issue earlier on. @sindresorhus let me know which PR you'd prefer to go with.

File: cli.js [hunk 1]
```diff
@@ -57,17 +63,22 @@ const speedText = () =>
 		downloadColor(downloadSpeed());
 
 const speed = () => speedText() + '\n\n';
+const getVerboseLog = () => `Latency: ${data.latency}${data.latencyUnit} (unloaded) ${data.bufferbloat}${data.bufferbloatUnit} (loaded)\nClient: ${data.client.location}  ${data.client.ip}  ${data.client.isp}\nServer(s): ${data.serverLocations}`;
```
Comment (itaisteinherz) [inline cli.js]: This should probably be split up to several template strings, or I could add newlines and strip indent using [`strip-indent`](https://github.com/sindresorhus/strip-indent). Let me know what you think is better.

**Comment by itaisteinherz** [comment]
@sindresorhus Friendly ping 🏓

**Comment by sindresorhus** [comment]
Sorry, I was waiting for https://github.com/sindresorhus/fast-cli/pull/40 as it was submitted first, but no response there, so it's all yours. I prefer the look of https://github.com/sindresorhus/fast-cli/pull/40 though, so if you could make it look like that and address the feedback given there, it would be great.

**Comment by sindresorhus** [comment]
@itaisteinherz Bump :)

**Comment by Telematica** [comment]
Any news on this feature? Would love having it!

**Comment by sindresorhus** [comment]
@itaisteinherz Last bump

**Comment by itaisteinherz** [comment]
@sindresorhus Thanks for pinging me, I'll get to it this evening and push my fixes.

**Comment by itaisteinherz** [comment]
@sindresorhus I fixed the comments you left [here](https://github.com/sindresorhus/fast-cli/pull/40#issuecomment-586744723).

File: cli.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ const cli = meow(`
 
 	Options
 	  --upload, -u  Measure upload speed in addition to download speed
+	  --verbose     Get verbose logging on latency and request metadata
```
Comment (sindresorhus) [inline cli.js:17]:
```suggestion
	  --verbose     Include info on latency and request metadata
```

File: cli.js [hunk 3]
```diff
@@ -57,17 +63,22 @@ const speedText = () =>
 		downloadColor(downloadSpeed());
 
 const speed = () => speedText() + '\n\n';
+const getVerboseLog = () => `   Latency:  ${data.latency}${data.latencyUnit} ${chalk.dim('(unloaded)')}  ${data.bufferbloat}${data.bufferbloatUnit} ${chalk.dim('(loaded)')}\n    Client:  ${data.client.location} ${data.client.ip} ${data.client.isp}\n Server(s):  ${data.serverLocations}`;
```
Comment (sindresorhus) [inline cli.js:66]:
```suggestion
const getVerboseLog = () => `   Latency:  ${data.latency}${data.latencyUnit} ${chalk.dim('(unloaded)')}  ${data.bufferbloat}${data.bufferbloatUnit} ${chalk.dim('(loaded)')}\n    Client:  ${data.client.location} ${data.client.ip} ${data.client.isp}\n Servers:  ${data.serverLocations}`;
```

**Comment by sindresorhus** [comment]
Latency should be shown when it's ready. Not just at the end.

**Comment by sindresorhus** [comment]
Can you fix the merge conflict? (You also need to make `--verbose` compatible with the `--single-line` flag. It doesn't have to be a single line, just remove all unneeded whitespace when `--single-line`.

**Comment by kirubakaran** [comment]
@itaisteinherz thanks for your work. Hoping to see this in the master soon!

File: cli.js (hunk 2 above)
Comment (itaisteinherz) [reply cli.js]: Done

File: cli.js (hunk 3 above)
Comment (itaisteinherz) [reply cli.js]: Done

File: cli.js [hunk 4]
```diff
@@ -62,21 +68,37 @@ const uploadColor = string => (data.isDone ? chalk.green(string) : chalk.cyan(st
...
+const latencyColor = string => (data.isLatencyDone ? chalk.white(string) : chalk.cyan(string));
+const bufferbloatColor = string => (data.isBufferbloatDone ? chalk.white(string) : chalk.cyan(string));
+
 const speedText = () =>
 	cli.flags.upload ?
 		`${downloadColor(downloadSpeed())} ${chalk.dim('/')} ${uploadColor(uploadSpeed())}` :
 		downloadColor(downloadSpeed());
 
-const speed = () => speedText() + lineBreak(2);
+const latencyText = () => `Latency:  ${latencyColor(data.latency + data.latencyUnit)} ${chalk.dim('(unloaded)')}  ${bufferbloatColor(data.bufferbloat + data.bufferbloatUnit)} ${chalk.dim('(loaded)')}`;
```
Comment (itaisteinherz) [inline cli.js:79]: @sindresorhus Is it problematic that we use `chalk.dim` even when stdout is not a TTY?

**Comment by itaisteinherz** [comment]
@sindresorhus I applied your suggestions, resolved the conflicts and added support for using `--verbose` and `--single-line` together.

File: cli.js (hunk 4 above)
Comment (sindresorhus) [reply cli.js:79]: No, Chalk handles that for us.

**Comment by sindresorhus** [comment]
There should not be an empty line between the verbose stuff:

```
     Latency:  10ms (unloaded)  112ms (loaded)

      Client:  Fredrikstad, NO 212.251.200.182 Telenor
     Servers:  Open Connect, Netflix
```

**Comment by sindresorhus** [comment]
Notice the unalignment here:

```
❯ node cli.js --verbose --single-line
100 Mbps ↓ / 28 Mbps ↑
Latency:  12ms (unloaded)  61ms (loaded)
Client:  Fredrikstad, NO 212.251.200.182 Telenor
Servers:  Open Connect, Netflix
```

**Comment by sindresorhus** [comment]
I think there should be one more empty line between the speed and `Latency:`.

**Comment by skywinder** [comment]
Nice update. looking forward to merge ❤️

**Comment by sindresorhus** [comment]
Closing in favor of #52

**Pull request closed without merging**